system/jlib/jevent.cpp

```diff
@@ -1757,6 +1821,513 @@ IEventVisitor* createDumpCSVEventVisitor(IBufferedSerialOutputStream& out)
+        {
+            patterns.insert(token);
+            return true;
+        }
+    };
+
+    struct FilterTerm : public CInterface
+    {
+        virtual bool matches(EventType evt, const char* value) const { return true; }
+        virtual bool matches(EventType evt, bool value) const { return true; }
+        virtual bool matches(EventType evt, __uint64 value) const { return true; }
+        virtual bool accept(const char* values)
+        {
+            StringArray tokens;
+            tokens.appendList(values, ",", true);
+            ForEachItemIn(i, tokens)
+            {
+                if (!acceptToken(tokens.item(i)))
+                    return false;
+            }
+            return true;
+        }
+        virtual bool acceptToken(const char* token) = 0;
+    };
+
+    struct StringFilterTerm : public FilterTerm
+    {
+        StringMatchHelper helper;
+        bool matches(EventType evt, const char* value) const override
+        {
+            return helper.matches(value);
+        }
+
+        bool acceptToken(const char* token) override
+        {
+            return helper.acceptToken(token);
+        }
+    };
+
+    struct BoolFilterTerm : public FilterTerm
+    {
+        bool acceptTrue{false};
+        bool acceptFalse{false};
+
+        bool matches(EventType evt, bool value) const override
+        {
+            return (value && acceptTrue) || (!value && acceptFalse);
+        }
+
+        bool acceptToken(const char* token) override
+        {
+            bool value = strToBool(token);
+            if (value && !acceptTrue)
+                acceptTrue = true;
+            if (!value && !acceptFalse)
+                acceptFalse = true;
+            return true;
+        }
+    };
+
+    struct UnsignedFilterTerm : public FilterTerm
+    {
+        using Accepted = std::set<std::pair<__uint64, __uint64>>;
+        Accepted accepted;
+
+        bool matches(EventType evt, __uint64 value) const override
+        {
+            for (const std::pair<__uint64, __uint64>& range : accepted)
+            {
+                if (value > range.second)
+                    return false;
+                if (range.first <= value && value <= range.second)
+                    return true;
+            }
+            return false;
+        }
+
+        bool acceptToken(const char* token) override
+        {
+            char* tmp = nullptr;
+            const char* delim = strchr(token, '-');
+            if (!delim)
+            {
+                __uint64 first = strtoull(token, &tmp, 0);
+                if (*tmp == '\0')
+                    return acceptRange(first, first);
+            }
+            else if (delim == token)
+            {
+                __uint64 last = strtoull(delim + 1, &tmp, 0);
+                if (*tmp == '\0')
+                    return acceptRange(0, last);
+            }
+            else
+            {
+                __uint64 first = strtoull(token, &tmp, 0);
```

Comment:
Code here could be simplified, but I'm not sure it is worth revisiting.   (e.g. Avoid most of the special casing, use variables for low and high, return false on error, and have a single call to add the range.)